src/types/bytes.rs

```diff
@@ -142,6 +186,15 @@ impl<I: SliceIndex<[u8]>> Index<I> for PyBytes {
     }
 }
 
+/// This is the same way [Vec] is indexed.
+impl<I: SliceIndex<[u8]>> Index<I> for Bound<'_, PyBytes> {
```

Comment:
we could additionally implement it on `&Bound` as well, so it also works an a borrowed bound

Reply:
I think it's not needed due to the way `bytes[I]` gets desugared into the `index` call.

I've pushed an additional part to `test_bound_bytes_index` which indexes on `&Bound<PyBytes>`, see if you agree with me that the one implementation is sufficient?

Reply:
Perfect, that even better! 👍